services/director/src/simcore_service_director/producer.py

```diff
@@ -135,118 +202,30 @@ def _get_swarm_network(client: DockerClient) -> docker.models.networks.Network:
-        main_service: bool):
-    # pylint: disable=C0103
-    # add the service uuid to the docker service
-    service_type = "main" if main_service else "dependency"
-    if "labels" in docker_service_runtime_parameters:
-        docker_service_runtime_parameters["labels"]["type"] = service_type
-    else:
-        docker_service_runtime_parameters["labels"] = {"type": service_type}
-    log.debug("Added type label to docker runtime parameters: %s",
-              docker_service_runtime_parameters["labels"])
-
-
-def __add_network_to_service_runtime_params(
-        docker_service_runtime_parameters: Dict,
-        docker_network: docker.models.networks.Network):
-
-    # pylint: disable=C0103
-    if "networks" in docker_service_runtime_parameters:
-        docker_service_runtime_parameters["networks"].append(docker_network.id)
-    else:
-        docker_service_runtime_parameters["networks"] = [docker_network.id]
-    log.debug("Added network parameter to docker runtime parameters: %s",
-              docker_service_runtime_parameters["networks"])
-
-
-def __add_env_variables_to_service_runtime_params(
-        docker_service_runtime_parameters: Dict,
-        user_id: str,
-        project_id: str,
-        node_uuid: str,
-        node_base_path: str):
-
-    service_env_variables = ["=".join([key, value]) for key, value in config.SERVICES_DEFAULT_ENVS.items()]
-    # add specifics
-    service_env_variables.append("=".join(["SIMCORE_USER_ID", user_id]))
-    service_env_variables.append("=".join(["SIMCORE_NODE_UUID", node_uuid]))
-    service_env_variables.append("=".join(["SIMCORE_PROJECT_ID", project_id]))
-    service_env_variables.append("=".join(["SIMCORE_NODE_BASEPATH", node_base_path or ""]))
-
-    if "env" in docker_service_runtime_parameters:
-        docker_service_runtime_parameters["env"].extend(service_env_variables)
-    else:
-        docker_service_runtime_parameters["env"] = service_env_variables
-    log.debug("Added env parameter to docker runtime parameters: %s", docker_service_runtime_parameters["env"])
-
-
-def _add_extra_hosts_to_service_runtime_params(docker_service_runtime_parameters: Dict):
-    log.debug("Getting extra hosts with suffix: %s",config.EXTRA_HOSTS_SUFFIX)
-    extra_hosts = get_system_extra_hosts(config.EXTRA_HOSTS_SUFFIX)
-    if "hosts" in docker_service_runtime_parameters:
-        docker_service_runtime_parameters["hosts"].update(extra_hosts)
-    else:
-        docker_service_runtime_parameters["hosts"] = extra_hosts
-
-
-def __set_service_name(docker_service_runtime_parameters: Dict,
-                       service_name: str,
-                       node_uuid: str):
-    # pylint: disable=C0103
-    docker_service_runtime_parameters["name"] = service_name + "_" + node_uuid
-    log.debug("Added service name parameter to docker runtime parameters: %s",
-              docker_service_runtime_parameters["name"])
-
+async def _get_swarm_network(client: aiodocker.docker.Docker) -> Dict:
```

Comment:
I am surprised that python likes the fact that you are overloading ``_get_swarm_network`` (one sync and the other async). I would rename them.

Reply:
it does not like it...
it was a bad merge